src/main/java/com/dougnoel/sentinel/webdrivers/WebDriverFactory.java

```diff
@@ -92,32 +94,57 @@ public static WebDriver instantiateWebDriver() {
+        	driver = instantiateRemoteWebDriver(browser, browserVersion, gridUrl);
+        }else {
+            // Initialize the driver object based on the browser and operating system (OS).
+            // Throw an error if the value isn't found.   	
+        	switch (browser) {
+            case "chrome":
+            	String headless = ConfigurationManager.getOptionalProperty("headless");
+            	if (headless == null || headless.equalsIgnoreCase("false"))
+            		driver = ChromeDriverFactory.createChromeDriver();
+            	else
+            		driver = ChromeDriverFactory.createHeadlessChromeDriver();
+                break;
+            case "firefox":
+            	driver = FirefoxDriverFactory.createFirefoxDriver();
+                break;
+            case "internetexplorer":
+            	driver = createInternetExplorerDriver();
+                break;
+            case "safari":
+            	driver = createSafariDriver();
+                break;
+            default:
+                throw new WebDriverException(SentinelStringUtils.format("Invalid browser type '{}' passed to WebDriverFactory. Could not resolve the reference. Check your spelling. Refer to the Javadoc for valid options.", browser));
+            }
+        }
 
-        String browser = ConfigurationManager.getBrowserName();
+        
 
-        // Initialize the driver object based on the browser and operating system (OS).
-        // Throw an error if the value isn't found.   	
-    	switch (browser) {
-        case "chrome":
-        	String headless = ConfigurationManager.getOptionalProperty("headless");
-        	if (headless == null || headless.equalsIgnoreCase("false"))
-        		driver = ChromeDriverFactory.createChromeDriver();
-        	else
-        		driver = ChromeDriverFactory.createHeadlessChromeDriver();
-            break;
-        case "firefox":
-        	driver = FirefoxDriverFactory.createFirefoxDriver();
-            break;
-        case "internetexplorer":
-        	driver = createInternetExplorerDriver();
-            break;
-        case "safari":
-        	driver = createSafariDriver();
-            break;
-        default:
-            throw new WebDriverException(SentinelStringUtils.format("Invalid browser type '{}' passed to WebDriverFactory. Could not resolve the reference. Check your spelling. Refer to the Javadoc for valid options.", browser));
-        }
+        return driver;
+    }
+    
+    private static WebDriver instantiateRemoteWebDriver(String browser, String browserVersion, String gridUrl) {
 
+    	DesiredCapabilities capability = new DesiredCapabilities();
+    	capability.setCapability("browserName", browser);
+    	capability.setCapability("browserVersion", browserVersion);
+    	
+        try {
+            driver = new RemoteWebDriver(new URL(gridUrl), capability);
+        } catch (Exception e) {
+            e.printStackTrace();
```

Comment:
If you need to write something to console, use the log4j2 logger. You can see examples of it in other files, or I can help you out.

Reply:
let me know about the above comment, based on that I can use log4j or remove try catch

Reply:
You don't need to log anything, just wrap and throw the exception.